IanGRap/Squish
# disabled vsync to improve performance

Goal:
- Reduce CPU usage by disabling VSync

IMPORTANT: 
- Once you switch to this branch, you must restart Unity to see the effect. At least, I had to.

What changed:
- I went into Edit -> Project Settings -> Quality -> Other -> V Sync Count and changed it to Don't Sync for every quality setting (Medium, High, Very High, Ultra)
- Before, VSync was enabled for every setting except low and very low. This is why when my computer wasn't plugged in, I was only able to push out decent performance on very low and low settings (I think)

How to Test:
- Go to the development branch and open the profiler from the Window tab
- Play the game and watch the CPU usage. For me, Initilization.PlayerUpdateTime -> WaitForTargetFPS was by far the highest usage (usually between 75 and 80%)
- Go to the optimization branch and restart Unity
- Play the game again and watch the profiler. The WaitForTargetFPS line should be gone. 
- For me, the next highest CPU usage was Gfx.WaitForPresent, which from what I can tell, indicates the amount of time the CPU is waiting for the GPU to finish processing the current frame. I only researched a little but it seems like it is not necessarily a problem. We should just monitor performance and consider it. 

Note:
- I was able to notice a difference in performance when I set my computer to use integrated graphics instead of my high powered graphics card. It is especially noticeable inside a standalone build for me. If you have a beastly rig, you may not notice a difference without looking at the profiler.

Before (the big yellow spike is vsync):
![image](https://user-images.githubusercontent.com/28333213/55674975-7b510700-5870-11e9-8b48-43ee9520c029.png)


After:
![image](https://user-images.githubusercontent.com/28333213/55674983-93288b00-5870-11e9-9c3a-5514ffeb9626.png)
